bindings/xmtp_rust_swift/src/types.rs

```diff
@@ -0,0 +1,122 @@
+use xmtp_proto::xmtp::message_api::v1::{
+    cursor as cursor_proto, Cursor, Envelope, IndexCursor, PagingInfo,
+    QueryResponse as ProtoQueryResponse, SortDirection,
+};
+
+impl From<crate::ffi::SortDirection> for SortDirection {
```

Comment:
This was super annoying, but Copilot was able to write most of these on the first shot.

Reply:
🤖 🙏